File: spring_boot_backend/src/main/java/com/spring_boot_backend/product/domain/Product.java
package com.spring_boot_backend.product.domain;


import lombok.*;

import java.util.HashSet;
import java.util.Set;


//Lombok annotations auto generating basic code
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor(access = AccessLevel.PUBLIC)
@EqualsAndHashCode(onlyExplicitlyIncluded = true)
public class Product{

    @EqualsAndHashCode.Include
    private String id = null;
    @NonNull
    private String productName;
    @NonNull
    private String description;
    @NonNull
    private Set<String> targetMarketStack = new HashSet<>();
    @NonNull
    private Set<String> technologiesStack = new HashSet<>();

    //No id constructor for inserting in the database where the id is auto generated
    public Product (@NonNull String productName, @NonNull String description,
                    @NonNull Set<String> targetMarketStack, @NonNull Set<String> technologiesStack){
        this.productName = productName;
        this.description = description;
        this.targetMarketStack = targetMarketStack;
        this.technologiesStack = technologiesStack;
    }
}

File: angular-frontend/src/app/shared/data-types/Product.ts
import {Optional} from "@angular/core";

export class Product{

  public id?: string;
  public productName: string;
  public description: string;
  public targetMarketStack: string[];
  public technologiesStack: string[];


  constructor( id: string | undefined, productName: string, description: string, targetMarketStack: string[], technologiesStack: string[]) {
    this.id = id;
    this.productName = productName;
    this.description = description;
    this.targetMarketStack = targetMarketStack;
    this.technologiesStack = technologiesStack;
  }

  public static createProductObject(id: string | undefined, productName: string, description: string, targetMarketStack: string, technologiesStack: string) : Product{
    let targetMarketStackTemp = targetMarketStack.split(',');
    let technologiesStackTemp = technologiesStack.split(',');
    return new Product(id, productName, description, targetMarketStackTemp, technologiesStackTemp);
  }
}
